src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2792,6 +2792,16 @@ private function registerWebhookConfiguration(array $config, ContainerBuilder $c
 
         $loader->load('webhook.php');
 
+        // If the Webhook is installed but the HttpClient or Serializer components are not available, we should throw an error
+        if (!ContainerBuilder::willBeAvailable('symfony/http-client', HttpClientInterface::class, ['symfony/framework-bundle'])) {
+            $container->register('webhook.transport', 'stdClass')
+                ->addError('You cannot use the "webhook transport" service since the HttpClient contracts are not installed. Try running "composer require symfony/http-client".');
+        }
+        if (!ContainerBuilder::willBeAvailable('symfony/serializer', Serializer::class, ['symfony/framework-bundle'])) {
+            $container->register('webhook.body_configurator.json', 'stdClass')
+                ->addError('You cannot use the "webhook JSON body configuration" service since the Serializer component is not installed. Try running "composer require symfony/serializer".');
```

Comment:
```suggestion
                ->addError('You cannot use the "webhook transport" service since the Serializer component is not installed. Try running "composer require symfony/serializer".');
```

but I think this needs more logic, similar to what we do for request_payload: the serializer can be installed but not enabled.